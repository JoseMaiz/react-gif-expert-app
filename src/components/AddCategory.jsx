import { useState } from 'react'
import Protypes from 'prop-types';


export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState("")

    const onInputChange = ({target})=>{
        setInputValue(target.value);
    }

    const onSubmit = (event)=>{
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;

        // * Forma importando la funcion del useState setCategories del parent component
        // setCategories((categ)=> [inputValue,... categ])

        // *Forma importando una funcion que tenga todo el proceso ya realizado de la actualizacion del estado del useStade "setCategories" del parent component 
        onNewCategory(inputValue.trim());
        setInputValue("");
        
    }

    return (
        <form onSubmit={onSubmit} aria-label='form'>
            <input 
                type="text" 
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />

        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: Protypes.func.isRequired,
};
