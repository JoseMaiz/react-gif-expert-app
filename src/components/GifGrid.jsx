import { GifItem } from "./GifItem";

import { useFetchGitfs } from "../hooks/useFetchGitfs";

export const GifGrid = ({category}) => {

    
    const {images, isLoading} = useFetchGitfs(category);

    return (
        <>
            <h3>{category}</h3>
            {
                // * Manera mas corta del condicionante para que ponga cargando...
                isLoading && (<h2>Cargando...</h2>) 
            }
            <div className="card-grid">
                {images.map((image) =>(
                    
                    <GifItem 
                    
                        key={image.id}
                        {... image}

                    />
                ))}
            </div>
        </>
    )
}
