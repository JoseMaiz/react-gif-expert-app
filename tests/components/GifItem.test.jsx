import {render, screen} from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';
import { getGifs } from '../../src/helpers/getGifs';

const image ={
    id: '977YesTjNfQC7vQiph',
    title: 'Dragon Ball GIF by Toei Animation',
    url: 'https://media0.giphy.com/media/977YesTjNfQC7vQiph/giphy.gif?cid=14246f2958zmvh8rnuckuhfxozejfzwy6v2r0p4qfswrp83i&ep=v1_gifs_search&rid=giphy.gif&ct=g'
}

describe('Pruebas en el GifItem', () => {
    test('Debe de hacer match con el snapshot', async () => {
        
        const {container} = render(<GifItem key ={image.id} {...image}/>)

        expect(container).toMatchSnapshot()

    });

    test('Debe de mostrar la imagen con el url y el ALT indicado',async () => {
        
        render(<GifItem key ={image.id} {...image}/>);

        // screen.debug();

        // *Manera de evaluar atributo por atribulo (No muy eficiente)
        // expect(screen.getByRole("img").src).toBe(image.url);
        // expect(screen.getByRole("img").alt).toBe(image.title);

        // *Manera mas eficiente y elegante de probar varios atributos. 
        const {src,alt} = screen.getByRole("img");
        expect(alt).toBe(image.title);
        expect(src).toBe(image.url);

    });

    test('Debe de mostrar el titulo en el componente', () => {
        
        render(<GifItem key ={image.id} {...image}/>);

        expect(screen.getByText(image.title)).toBeTruthy();
    })

})