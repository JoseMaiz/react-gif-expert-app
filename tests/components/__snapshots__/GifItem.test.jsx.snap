// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball GIF by Toei Animation"
      src="https://media0.giphy.com/media/977YesTjNfQC7vQiph/giphy.gif?cid=14246f2958zmvh8rnuckuhfxozejfzwy6v2r0p4qfswrp83i&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Dragon Ball GIF by Toei Animation
    </p>
  </div>
</div>
`;
